### PR TITLE
Implement annotation layer regression testing

### LINKED ISSUE #6673: Regression tests do not display annotations
In the annotation flags refactoring PR, @Snuffleupagus noticed that our regression testing framework does not actually display the annotations and therefore they are not compared in the snapshots. This means that a file like `annotation-border-styles.pdf` is tested without rendered annotations, which defeats the purpose of that test case.

The cause is probably that some parts of the annotation code are placed in the viewer, which the regression testing framework does not take into account. We might need to do something similar to PR #6619 to fix this.

### PR DESCRIPTION
**Pull request by timvandermeij**
Fixes #6673.

### CODE REVIEW

File: test/driver.js [hunk 1]
```diff
@@ -343,8 +433,35 @@ var Driver = (function DriverClosure() {
...
+                // The annotation builder will draw its content on the canvas.
+                initPromise =
+                  page.getAnnotations({ intent: 'display' }).then(
+                    function(annotations) {
+                      return rasterizeAnnotationLayer(annotationLayerContext,
+                                                      viewport, annotations,
+                                                      page);
+                  });
+              } else {
+                initPromise = Promise.resolve();
```
Comment (yurydelendik) [inline test/driver.js]: make `annotationLayerCanvas = null`; here

**Comment by timvandermeij** [comment]
I re-pushed the commit to remove one `"annotations": true` entry from the test manifest because that is a PDF with only printable annotations, which we do not render for the tests, so having it has no effect. Nothing else has changed since the last test run.

File: test/driver.js [hunk 2]
```diff
@@ -90,6 +113,109 @@ var rasterizeTextLayer = (function rasterizeTextLayerClosure() {
...
+      var div = document.createElement('div');
+      div.className = 'annotationLayer';
+
+      // Rendering annotation layer as HTML.
+      stylePromise.then(function (styles) {
+        style.textContent = styles;
+
+        var annotation_viewport = viewport.clone({ dontFlip: true });
+        PDFJS.AnnotationLayer.render(annotation_viewport, div, annotations,
+                                     page, new LinkServiceMock());
```
Comment (Snuffleupagus) [inline test/driver.js]: Just a reminder: With PR #6770 landed, the parameters now need to be passed in as an object instead.

Comment (timvandermeij) [inline test/driver.js]: Done in the new commit.

File: test/driver.js (hunk 1 above)
Comment (timvandermeij) [inline test/driver.js]: Done in the new commit.

**Comment by yurydelendik** [comment]
Looks good. Thank you for the patch.

/botio makeref

**Pull request merged**